lib/capistrano/s3/publisher.rb

```diff
@@ -110,7 +112,7 @@ def self.put_object(s3, bucket, target_path, path, file, only_gzip, extra_option
           mime_type = mime_type_for_file(base_name)
           options   = {
             :bucket => bucket,
-            :key    => File.join(target_path, path),
+            :key    => self.add_prefix(path, prefix: target_path),
```

Comment:
No need to write `self`. Per [RSG](https://github.com/bbatsov/ruby-style-guide/blob/master/README.md#class-and-self)